packages/flutter_tools/lib/src/base/utils.dart

```diff
@@ -350,7 +350,7 @@ List<String> _wrapTextAsLines(String text, { int start = 0, int columnWidth, @re
   // reconstitute the original string. This is useful for manipulating "visible"
   // characters in the presence of ANSI control codes.
   List<_AnsiRun> splitWithCodes(String input) {
-    final RegExp characterOrCode = RegExp('(\u001b\[[0-9;]*m|.)', multiLine: true);
+    final RegExp characterOrCode = RegExp('(\u001b[[0-9;]*m|.)', multiLine: true);
```

Comment:
Here I'm not sure what it should be (so I left the same semantic)

```dart
final RegExp characterOrCode = RegExp('(\u001b[[0-9;]*m|.)', multiLine: true);
final RegExp characterOrCode = RegExp('(\\u001b[[0-9;]*m|.)', multiLine: true);
final RegExp characterOrCode = RegExp('(\u001b\\[[0-9;]*m|.)', multiLine: true);
final RegExp characterOrCode = RegExp('(\\u001b\\[[0-9;]*m|.)', multiLine: true);
```

Reply:
Don't you have to escape a `[` in a RegExp?

Reply:
Also in general: I wonder if we should have a lint requiring raw strings for regexps.

Reply:
We probably can't require them since some appear to need interpolation, but otherwise, yeah.

Reply:
You could work around the lint by explicitly creating an interpolated string before the regexp - but I think it's more common to make a mistake with string escaping in a regex than to actually use an interpolated string correctly.

Reply:
>Don't you have to escape a [ in a RegExp?

That's why I pointed this line. The current regex value looks strange but I don't have enough knowledge to know if it is intentional.

>Also in general: I wonder if we should have a lint requiring raw strings for regexps.

I don't really agree. I even have a lint in mind (which will go against your proposal of raw string in regexp) to only use raw string if necessary. For instance I have no problem to have `RegExp('ab')`. As @zanderso mentionned interpolation (and other escape sequences) prevents always requiring a raw string.

I've other lints about strings in mind. One could help, it would advice to change string literal with only `\\` (as escape sequence) to a raw string.

Reply:
The regex is trying to match strings like we have here:
https://github.com/flutter/flutter/blob/e4014a55c87236eee3bbd4e491cd1e969c11c6b4/packages/flutter_tools/lib/src/base/terminal.dart#L96

Reply:
I think the good regex should then be `'(\u001b\\[[0-9;]*m|.)'`. This pattern will not match `A[1[1m` unlike `'(\u001b[[0-9;]*m|.)'`.

PR updated.